pkg/cli/admin/upgrade/status/controlplane.go

```diff
@@ -0,0 +1,105 @@
+package status
+
+import (
+	"io"
+	"strings"
+	"text/template"
+	"time"
+
+	v1 "github.com/openshift/api/config/v1"
+)
+
+type assessmentState string
+
+const (
+	assessmentStateProgressing assessmentState = "Progressing"
+	assessmentStateCompleted   assessmentState = "Completed"
+)
+
+type operators struct {
+	Total       int
+	Available   int
+	Progressing int
+	Degraded    int
+}
+
+type controlPlaneStatusDisplayData struct {
+	Assessment assessmentState
+	Completion float64
+	Duration   time.Duration
+	Operators  operators
+}
+
+func assessControlPlaneStatus(cv *v1.ClusterVersion, operators []v1.ClusterOperator, at time.Time) controlPlaneStatusDisplayData {
+	var displayData controlPlaneStatusDisplayData
+	var completed int
+
+	targetVersion := cv.Status.Desired.Version
+
+	for _, operator := range operators {
+		var isPlatformOperator bool
+		for annotation := range operator.Annotations {
+			if strings.HasPrefix(annotation, "exclude.release.openshift.io/") ||
+				strings.HasPrefix(annotation, "include.release.openshift.io/") {
+				isPlatformOperator = true
+				break
+			}
+		}
+		if !isPlatformOperator {
+			continue
+		}
+
+		var updated bool
+		for _, version := range operator.Status.Versions {
+			if version.Name == "operator" && version.Version == targetVersion {
+				updated = true
+				completed++
+				break
+			}
+		}
+		displayData.Operators.Total++
+		for _, condition := range operator.Status.Conditions {
+			switch {
+			case condition.Type == v1.OperatorAvailable:
+				if condition.Status == v1.ConditionTrue {
+					displayData.Operators.Available++
+				}
+			case condition.Type == v1.OperatorProgressing && condition.Status == v1.ConditionTrue && !updated:
```

Comment:
nit: `Available` splits  this into `case` and an `if`, but  here and for `Degraded` you check both `Type` and `Status` in the `case`.  Not a big deal, but you may want to consolidate around one or the other approach, just to make it slightly easier code-reading.

Reply:
Good catch, thanks!